tree/ntuple/v7/inc/ROOT/RNTuple.hxx

```diff
@@ -0,0 +1,236 @@
+      for (auto& value : *entry) {
+         value.GetField()->Read(index, &value);
+      }
+   }
+
+   RNTupleViewRange GetViewRange() { return RNTupleViewRange(0, fNEntries); }
+
+   /// Provides access to an individual field that can contain either a skalar value or a collection, e.g.
+   /// GetView<double>("particles.pt") or GetView<std::vector<double>>("particle").  It can as well be the index
+   /// field of a collection itself, like GetView<NTupleSize_t>("particle")
+   template <typename T>
+   RNTupleView<T> GetView(std::string_view fieldName) { return RNTupleView<T>(fieldName, fSource.get()); }
+   RNTupleViewCollection GetViewCollection(std::string_view fieldName) {
+      return RNTupleViewCollection(fieldName, fSource.get());
+   }
+
+   RIterator begin() { return RIterator(0); }
+   RIterator end() { return RIterator(fNEntries); }
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleWriter
+\ingroup NTuple
+\brief An RNTuple that gets filled with entries (data) and writes them to storage
+
+An output ntuple can be filled with entries. The caller has to make sure that the data that gets filled into an ntuple
+is not modified for the time of the Fill() call. The fill call serializes the C++ object into the column format and
+writes data into the corresponding column page buffers.  Writing of the buffers to storage is deferred and can be
+triggered by Flush() or by destructing the ntuple.  On I/O errors, an exception is thrown.
+*/
+// clang-format on
+class RNTupleWriter : public Detail::RNTuple {
+private:
+   static constexpr NTupleSize_t kDefaultClusterSizeEntries = 8192;
+   std::unique_ptr<Detail::RPageSink> fSink;
+   NTupleSize_t fClusterSizeEntries;
+   NTupleSize_t fLastCommitted;
+
+public:
+   static std::unique_ptr<RNTupleWriter> Recreate(std::unique_ptr<RNTupleModel> model,
+                                                  std::string_view ntupleName,
+                                                  std::string_view storage);
+   RNTupleWriter(std::unique_ptr<RNTupleModel> model, std::unique_ptr<Detail::RPageSink> sink);
+   RNTupleWriter(const RNTupleWriter&) = delete;
+   RNTupleWriter& operator=(const RNTupleWriter&) = delete;
+   ~RNTupleWriter();
+
+   /// The simplest user interface if the default entry that comes with the ntuple model is used
+   void Fill() { Fill(fModel->GetDefaultEntry()); }
+   /// Multiple entries can have been instantiated from the tnuple model.  This method will perform
+   /// a light check whether the entry comes from the ntuple's own model
+   void Fill(REntry *entry) {
+      for (auto& value : *entry) {
+         value.GetField()->Append(value);
+      }
+      fNEntries++;
+      if ((fNEntries % fClusterSizeEntries) == 0) CommitCluster();
+   }
+   /// Ensure that the data from the so far seen Fill calls has been written to storage
+   void CommitCluster();
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RCollectionNTuple
```

Comment:
I read this name incorrectly the first time.  I thought it would be a collection of RNTuples (i.e. a TChain .. )